src/twisted/test/test_defer.py

```diff
@@ -645,6 +645,150 @@ def test_cancelGatherResultsWithAllDeferredsCallback(self):
         self.assertEqual(callbackResult[1], "Callback Result")
 
 
+    def test_extractCurrentResult(self):
+        """
+        L{defer.extractCurrentResult} extracts the results from a Deferred that
+        has already had its callback function called.
+        """
+        result = object()
+
+        d = defer.Deferred()
+        d.callback(result)
+
+        self.assertIs(defer.extractCurrentResult(d), result)
+
+
+    def test_extractCurrentResultRaisesFailureException(self):
+        """
+        L{defer.extractCurrentResult} raises the original exception if the
+        result of the deferred is a failure.
+        """
+        d = defer.Deferred()
+        d.errback(ValueError())
+
+        self.assertRaises(ValueError, defer.extractCurrentResult, d)
+
+
+    def test_extractCurrentResultReturnsFailureWithNoRaises(self):
+        """
+        L{deferextractCurrentResult} returns the failure instance if
+        raises=False.
+        """
+        d = defer.Deferred()
+        d.errback(ValueError())
+
+        result = defer.extractCurrentResult(d, raises=False)
+
+        self.assertIsInstance(result, failure.Failure)
+        self.assertIsInstance(result.value, ValueError)
+
+
+    def test_extractCurrentResultWithoutResultsNotCaughtByException(self):
+        """
+        L{defer.extractCurrentResult} raises a L{defer.NoCurrentResult}
+        exception which is not caught when catching L{Exception}.
+        """
+        def inner(d):
+            try:
+                defer.extractCurrentResult(d)
+            except Exception:
+                return "error"
+            return "success"
+
+        d = defer.Deferred()
+        self.assertRaises(defer.NoCurrentResult, inner, d)
+
+        # In order to get 100% patch coverage we need to excerise the pass
+        # statement. This is a bit silly because we're basically testing our
```

Comment:
This talk about a "pass statement" which is confusing.  I think it must mean the `return "error"` statement which maybe used to just be `pass`?

I don't know that calling this silly is helping anyone (except maybe you, to blow off steam after having to write some code you thought was pointless to satisfy a policy you disagree with).  Just explaining why you run through the code twice here seems sufficient.